server/src/main/scala/actors/minutes/MinutesActorLike.scala

```diff
@@ -115,26 +149,30 @@ abstract class MinutesActorLike[A, B <: WithTimeAccessor](now: () => SDateLike,
     }
 
   def updateByTerminalDayAndGetDiff(container: MinutesContainer[A, B]): Future[Option[MinutesContainer[A, B]]] = {
-    val eventualUpdatedMinutesDiff = groupByTerminalAndDay(container)
-      .map {
-        case ((terminal, day), terminalDayMinutes) => handleUpdateAndGetDiff(terminal, day, terminalDayMinutes)
-      }
-    combineEventualMinutesContainers(eventualUpdatedMinutesDiff).map(Option(_))
+    val eventualUpdatedMinutesDiff = Source(groupByTerminalAndDay(container)).mapAsync(1) {
+      case ((terminal, day), terminalDayMinutes) => handleUpdateAndGetDiff(terminal, day, terminalDayMinutes)
+    }
+    combineEventualMinutesContainersStream(eventualUpdatedMinutesDiff).map(Option(_))
   }
 
   def groupByTerminalAndDay(container: MinutesContainer[A, B]): Map[(Terminal, SDateLike), Iterable[MinuteLike[A, B]]] =
     container.minutes
       .groupBy(simMin => (simMin.terminal, SDate(simMin.minute).getUtcLastMidnight))
 
-  private def combineEventualMinutesContainers(eventualUpdatedMinutesDiff: Iterable[Future[MinutesContainer[A, B]]]): Future[MinutesContainer[A, B]] =
-    Future
-      .sequence(eventualUpdatedMinutesDiff)
-      .map(_.foldLeft(MinutesContainer.empty[A, B])(_ ++ _))
+  private def combineEventualMinutesContainersStream(eventualUpdatedMinutesDiff: Source[MinutesContainer[A, B], NotUsed]): Future[MinutesContainer[A, B]] = {
```

Comment:
we could break a few of these functions into a location that would make them individually testable.